### PR TITLE
Mark some methods as protected to make it easier to write own SslCont…

### PR DESCRIPTION
**Pull request by normanmaurer**
…ext implementations

Motivation:

We should expose some methods as protected to make it easier to write custom SslContext implementations.
This will be reused by the code for https://github.com/netty/netty-incubator-codec-quic/issues/97

Modifications:

- Add protected to some static methods which are useful for sub-classes
- Remove some unused methods
- Move *Wrapper classes to util package and make these public

Result:

Easier to write custom SslContext implementations

### CODE REVIEW

File: handler/src/main/java/io/netty/handler/ssl/SslContext.java [hunk 1]
```diff
@@ -1157,7 +1159,7 @@ private static PrivateKey getPrivateKeyFromByteBuffer(ByteBuf encodedKeyBuf, Str
     protected static TrustManagerFactory buildTrustManagerFactory(
             File certChainFile, TrustManagerFactory trustManagerFactory)
             throws NoSuchAlgorithmException, CertificateException, KeyStoreException, IOException {
-        return buildTrustManagerFactory(certChainFile, trustManagerFactory, KeyStore.getDefaultType());
```
Comment (chrisvest) [inline handler/src/main/java/io/netty/handler/ssl/SslContext.java:1160]: Why this change?

**Review by chrisvest — COMMENTED** [review]
Just one thing I don't understand.

File: handler/src/main/java/io/netty/handler/ssl/SslContext.java (hunk 1 above)
Comment (normanmaurer) [reply handler/src/main/java/io/netty/handler/ssl/SslContext.java:1160]: because we call `KeyStore.getDefaultType()` in the method itself if `null` is passed

**Review by chrisvest — APPROVED** [review]
